sklearn/base.py

```diff
@@ -449,6 +449,25 @@ def fit_transform(self, X, y=None, **fit_params):
             return self.fit(X, y, **fit_params).transform(X)
 
 
+class DensityMixin(object):
+    """Mixin class for all density estimators in scikit-learn."""
+    _estimator_type = "DensityEstimator"
+
+    def score(self, X, y = None):
+        """Returns the log likelihood under the model on the given data X.
+
+        Parameters
+        ----------
+        X : array-like, shape = (n_samples, n_features)
+
+        Returns
+        -------
+        score: float
+            sum of log probability of samples
+        """
+        return np.sum(self.score_samples(X))
```

Comment:
I agree. This ```score``` could be different from the log-likelihood of the model on the training data. If we use ```mean```, we should let users know that, and provide another method to return the log-likelihood.